packages/flutter_tools/lib/src/commands/build_windows.dart

```diff
@@ -52,10 +61,26 @@ class BuildWindowsCommand extends BuildSubCommand {
     if (!globals.platform.isWindows) {
       throwToolExit('"build windows" only supported on Windows hosts.');
     }
+
+    String? targetPlatformArg = stringArg('target-platform');
+    final String channel = globals.flutterVersion.channel;
+    final bool onStableOrBetaChannel = channel == 'stable' || channel == 'beta';
+    if ((targetPlatformArg != null) && onStableOrBetaChannel) {
+      throwToolExit('Option --target-platform can only be used on master channel.');
+    }
```

Comment:
Please add a test that checks that `--target-platform` fails if the channel is `stable` or `beta`. We want to make sure we don't regress this :)

Reply:
We'll do, once all other issues are solved :+1:.

Reply:
After updating this PR, this option is not present anymore. Thus, there is no test associated.